<?xml version="1.0"?>
<launch>

  <arg name="robot_ip" default="127.0.0.1"/>
  <arg name="plotting" default="true"/>
  <arg name="rviz" default="false"/>
  <arg name="paused" default="true" />
  <arg name="use_sim_time" default="true"/>
  <arg name="debug" default="true"/> <!-- Включить для вывода результатов расчетов -->
  <arg name="connect_robot" default="true"/> <!-- Подключение к роботу (в симуляции или к реальному роботу) -->
  <arg name="delay" default="0.5" />  <!-- Задержка для запуска ноды -->
  <arg name="prefix" default="ur5_"/>
  <arg name="rvizconfig" default="$(find ur5_husky_scene)/config/rviz/trajopt.rviz" />
  <arg name="vx300s" default="true"/>
  <arg name="gazebo" default="true"/>
  <arg name="script" default=""/> <!-- Название сценария -->
  <arg name="use_robot" default="false"/> <!-- Использовать URSim или UR5 -->
  <arg name="rosbag" default="false" /> <!-- True, если планируется проигрывание сенсорной информации из росбэга -->
  <arg name="delay_loop_rate" default="50.0" /> <!-- Любое положительное число. Не может быть 0! -->
  <arg name="ur_speed" default="0.1" />
  <arg name="ur_acceleration" default="10.0" />
  <arg name="ur_blend" default="0.0" />
  
  <!-- Варианты сценариев: open_lock (открыть ящик тумбы) -->

  <!-- Load controller configuration -->
  <rosparam command="load" file="$(find husky_control)/config/control.yaml" />
  <rosparam command="load" file="$(find ur5_husky_scene)/config/arm_controller_ur5.yaml" />
  <rosparam command="load" file="$(find ur5_husky_scene)/config/vx300s_trajectory_controllers.yaml"/>

  <!-- Load robot description -->
  <include file="$(find ur5_husky_scene)/launch/robot_load_ur5.launch" >
    <arg name="prefix" value="$(arg prefix)" />
  </include>

  <node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher" />

  <!-- Проба для Gazebo -->
  <include file="$(find gazebo_ros)/launch/empty_world.launch" if="$(arg gazebo)">
    <arg name="world_name" value="$(find ur5_husky_scene)/worlds/scene.world"/>
    <arg name="paused" value="false"/>
    <arg name="use_sim_time" value="true"/>
    <arg name="gui" value="true"/>
    <arg name="headless" value="false"/>
    <arg name="debug" value="false"/>
  </include>

  <!-- push robot_description to factory and spawn robot in gazebo -->
  <node name="spawn_gazebo_model" pkg="gazebo_ros" type="spawn_model" 
        args="-urdf -param robot_description -model robot -z 0.2" respawn="false" output="screen" if="$(arg gazebo)"/>

  <!-- Start joint state controller -->
  <node name="arm_controller_spawner" pkg="controller_manager" type="controller_manager" 
        args="spawn ur5_controller" respawn="false" output="screen"/>

  <!-- Husky -->

  <!-- Load controller configuration -->
  <rosparam command="load" file="$(find husky_control)/config/control.yaml" />

  <!-- Spawn controllers -->
  <!-- <node name="base_controller_spawner" pkg="controller_manager" type="controller_manager"
        args="spawn husky_velocity_controller" respawn="false" output="screen"/> -->

</launch>